config.go

```diff
@@ -516,92 +513,76 @@ func loadConfig() (*config, []string, error) {
 	case chaincfg.SimNetParams().Name:
 		cfg.net = &simNetParams
 	default:
-		err := fmt.Errorf("unknown network provided: %v", cfg.ActiveNet)
-		fmt.Fprintln(os.Stderr, err)
-		fmt.Fprintln(os.Stderr, usageMessage)
+		err := fmt.Errorf("%s: unknown network provided: %v", funcName,
+			cfg.ActiveNet)
 		return nil, nil, err
 	}
 
 	// Use separate data and log directories for each Decred network.
 	cfg.DataDir = cleanAndExpandPath(filepath.Join(cfg.DataDir, cfg.net.Name))
 	cfg.LogDir = cleanAndExpandPath(filepath.Join(cfg.LogDir, cfg.net.Name))
 
-	logRotator = nil
-
 	// Initialize log rotation.  After log rotation has been initialized, the
 	// logger variables may be used.
 	initLogRotator(filepath.Join(cfg.LogDir, defaultLogFilename))
 
+	// Special show command to list supported subsystems and exit.
+	if cfg.DebugLevel == "show" {
+		fmt.Println("Supported subsystems", supportedSubsystems())
+		os.Exit(0)
+	}
+
+	// Parse, validate, and set debug log level(s).
+	if err := parseAndSetDebugLevels(cfg.DebugLevel); err != nil {
+		err := fmt.Errorf("%s: %w", funcName, err)
+		return nil, nil, err
+	}
+
 	// Ensure the admin password is set.
 	if cfg.AdminPass == "" {
-		err := fmt.Errorf("the adminpass option is not set")
-		fmt.Fprintln(os.Stderr, err)
-		fmt.Fprintln(os.Stderr, usageMessage)
+		err := fmt.Errorf("%s: the adminpass option is not set", funcName)
 		return nil, nil, err
 	}
 
 	// Ensure the dcrd rpc username is set.
 	if cfg.RPCUser == "" {
-		err := fmt.Errorf("the rpcuser option is not set")
-		fmt.Fprintln(os.Stderr, err)
-		fmt.Fprintln(os.Stderr, usageMessage)
+		err := fmt.Errorf("%s: the rpcuser option is not set", funcName)
 		return nil, nil, err
 	}
 
 	// Ensure the dcrd rpc password is set.
 	if cfg.RPCPass == "" {
-		err := fmt.Errorf("the rpcpass option is not set")
-		fmt.Fprintln(os.Stderr, err)
-		fmt.Fprintln(os.Stderr, usageMessage)
+		err := fmt.Errorf("%s: the rpcpass option is not set", funcName)
 		return nil, nil, err
 	}
 
 	// Create the data directory.
 	err = os.MkdirAll(cfg.DataDir, 0700)
 	if err != nil {
-		str := "%s: unable to create data directory (%s): %v"
+		const str = "%s: unable to create data directory (%s): %w"
 		err := fmt.Errorf(str, funcName, cfg.DataDir, err)
-		fmt.Fprintln(os.Stderr, err)
 		return nil, nil, err
```

Comment:
This wasnt showing usage previously, now it is.

Reply:
Good point.  I was mainly just porting over from dcrd and failure to create the default config file there is just a normal `fmt.Fprintf` instead of an error like it is here.  I missed that it would indeed show the usage in that case.

I suspect it probably should actually be like dcrd and just log an error and move on since it isn't required to exist in order to run.

At any rate, I've modified to suppress it for this PR and we can change the aforementioned behavior in a separate PR if desired.